src/python_framework_v02/troute/nhd_io.py

```diff
@@ -161,6 +163,39 @@ def get_ql_from_wrf_hydro(qlat_files, index_col="station_id", value_col="q_later
 
     return ql
 
+def get_usgs_from_wrf_hydro(routelink_file,usgs_timeslices_folder,file_name):
+
+    with xr.open_dataset("../../test/input/geo/routelink/routeLink_subset.nc") as ds:
+        df = ds.to_dataframe()
+        df = df.reset_index()
+        df.drop(df.columns.difference(['link','to','gages','ascendingIndex']), 1, inplace=True)
```

Comment:
I think this block can be greatly simplified. This drop statement, for example, can be avoided by selecting the desired columns before creating the dataframe. This avoids loading everything in the netcdf file into memory. The conversions to strings and nan replacement steps are also unnecessary. If we want a float column, then we can convert to a float directly and coerce failures to nan.
```python
df = ds[['link', 'to', 'gages', 'ascendingIndex']].to_dataframe().reset_index()
df['gages'] = pd.to_numeric(df['gages'], errors='coerce')
df = df.loc[pd.notna(df['gages'])]
df = df.set_index('gages')
```

Reply:
This is still producing NaN values.